### DISCUSSION TITLE
What is vaex logic to define start of bins with BinnerTime?

### DISCUSSION THREAD

**Discussion by yohplala** [vaexio/vaex#1965] [Q&A]
Hi,
I am surprised by the start timestamp retained by vaex for binning this dataframe.
```python
import vaex
import pandas as pd
ts = pd.DatetimeIndex(map(pd.Timestamp,  ["2022/03/01 00:00", "2022/03/01 05:00", "2022/03/01 08:00"]))
vdf = vaex.from_pandas(pd.DataFrame({"ts":ts, "val":range(len(ts))}))
vdf
Out[15]: 
  #  ts                               val
  0  2022-03-01 00:00:00.000000000      0
  1  2022-03-01 05:00:00.000000000      1
  2  2022-03-01 08:00:00.000000000      2

vdf.groupby(vaex.BinnerTime(vdf.ts, resolution='h', every=4), agg={'sum': vaex.agg.sum("val")})
Out[16]: 
  #  ts               sum
  0  2022-02-28 23      0
  1  2022-03-01 03      1
  2  2022-03-01 07      2
```
Please, what is vaex logic for defining `2022-02-28 23` as the timestamp for the 1st bin?

Pandas will take as timestamp of the 1st bin, the 1st timestamp in seed dataframe, in this case `2022-03-01 00`.
(but this behavior can be controlled with `origin` parameter as highlighted in #1959)

Without having #1959, understanding vaex logic might offer me a workaround to still work with vaex for such aggregation. I could then 'force' a suitable timestamp so as having the 1st bin starting when I want it to start.

Thanks for your help.
Bests

**Comment by yohplala** [comment]
Hi,
Please, any idea? Thanks a lot for your help. Bests.

**Reply by maartenbreddels** [reply]
It seems be related to this difference:
<img width="606" alt="image" src="https://user-images.githubusercontent.com/1765949/162712677-9b1cac20-321c-4657-a84d-12826114f14c.png">


Which should be fixed by https://github.com/vaexio/vaex/pull/1469
